js/framework/historyStore.js

```diff
@@ -319,12 +319,16 @@ var HistoryStore = new Lang.Class({
                     model: model,
                     context_label: model.title,
                 });
-            } else if (model instanceof DModel.Media) {
+            } else if (model instanceof DModel.Media || model instanceof DModel.Audio) {
                 let old_item = this.get_current_item();
+                let context = old_item.model ? old_item.model.resources : null;
+                if (!context || context.length === 0) {
+                    context = [model];
+                }
```

Comment:
What does this case cover? It reminds me of *load_dbus_item* in *CourseHistoryStore*, which sets the context the to the selected model as in this case. Is it possible that that covers the case you're thinking of here?

Relatedly, you may want to change that function to include DModel.Audio as well:

https://github.com/endlessm/eos-knowledge-lib/blob/5743a58f434e0642efc55309d8d10773dcc9826d/js/framework/courseHistoryStore.js#L91

Reply:
The problem is, I'm not sure if `CourseHistoryStore` is suitable since this is not necessarily a course since it doesn't use that template, it looks more wrong than using this fallback.